internal/jobcontainers/jobcontainer.go

```diff
@@ -0,0 +1,424 @@
+// is mounted this call will just return nil.
+func (c *JobContainer) Release(ctx context.Context) error {
+	c.m.Lock()
+	defer c.m.Unlock()
+
+	log.G(ctx).WithFields(logrus.Fields{
+		"id":   c.id,
+		"path": c.sandboxMount,
+	}).Warn("removing sandbox volume mount")
+
+	if c.sandboxMount != "" {
+		if err := removeSandboxMountPoint(ctx, c.sandboxMount); err != nil {
+			return errors.Wrap(err, "failed to remove sandbox volume mount path")
+		}
+		if err := layers.UnmountContainerLayers(ctx, c.spec.Windows.LayerFolders, "", nil, layers.UnmountOperationAll); err != nil {
+			return errors.Wrap(err, "failed to unmount container layers")
+		}
+		c.sandboxMount = ""
+	}
+	return nil
+}
+
+// Start starts the container. There's nothing to "start" for job containers, so this just
+// sets the start timestamp.
+func (c *JobContainer) Start(ctx context.Context) error {
+	c.startTimestamp = time.Now()
+	return nil
+}
+
+// Close closes any open handles.
+func (c *JobContainer) Close() error {
+	if err := c.job.Close(); err != nil {
+		return err
+	}
+	c.closedWaitOnce.Do(func() {
+		c.waitError = hcs.ErrAlreadyClosed
+		close(c.waitBlock)
+	})
+	return nil
+}
+
+// ID returns the ID of the container. This is the name used to create the job object.
+func (c *JobContainer) ID() string {
+	return c.id
+}
+
+// Shutdown gracefully shuts down the container.
+func (c *JobContainer) Shutdown(ctx context.Context) error {
+	log.G(ctx).WithField("id", c.id).Debug("shutting down job container")
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*5)
+	defer cancel()
+	return c.shutdown(ctx)
+}
+
+// shutdown will loop through all the pids in the container and send a signal to exit.
+// If there are no processes in the container it will early return nil.
+// If the all processes exited message is not received within the context timeout set, it will
+// terminate the job.
+func (c *JobContainer) shutdown(ctx context.Context) error {
+	pids, err := c.job.Pids()
+	if err != nil {
+		return errors.Wrap(err, "failed to get pids in container")
+	}
+
+	if len(pids) == 0 {
+		return nil
+	}
+
+	for _, pid := range pids {
+		// If any process can't be signaled just wait until the timeout hits
+		if err := signalProcess(pid, windows.CTRL_SHUTDOWN_EVENT); err != nil {
+			log.G(ctx).WithField("pid", pid).Error("failed to signal process in job container")
+		}
+	}
+
+	select {
+	case <-c.exited:
+	case <-ctx.Done():
+		return c.Terminate(ctx)
+	}
+	return nil
+}
+
+// PropertiesV2 returns properties relating to the job container. This is an HCS construct but
+// to adhere to the interface for containers on Windows it is partially implemented. The only
+// supported property is schema2.PTStatistics.
+func (c *JobContainer) PropertiesV2(ctx context.Context, types ...hcsschema.PropertyType) (*hcsschema.Properties, error) {
+	return nil, errors.New("`PropertiesV2` call is not implemented for job containers")
```

Comment:
Do we expect to need to support metrics for k8s?

Reply:
Yes, I don't think for Alpha but I have the work pretty much done but didn't wanna stall this PR for that. Just need to find a way to get working set